src/scene/mod.rs

```diff
@@ -176,9 +176,12 @@ impl<'a> Node<'a> {
             }
         } else {
             Transform::Decomposed {
-                translation: self.json.translation,
-                rotation: self.json.rotation.0,
-                scale: self.json.scale,
+                translation: self.json.translation.unwrap_or_else(
```

Comment:
I suppose this is to make it easier to specify default transformation properties? If so, consider making a new type and implementing `Default::default` on it. Then the user can use `Foo { bar: 123, .. Default::default() }` to populate the rest of the struct with default values.

Reply:
Short answer, to follow the glTF 2.0 specification.

Longer answer, this change is a result of me changing the TRS properties to be optional to follow the specification. There's also a practical reason for this change, to allow exporting a node with the `.matrix` property present since the TRS properties and the `.matrix` properties are mutually exclusive. And any combination of `.translation`, `.rotation` and `.scale` is allowed to be present, if `.matrix` is not.

> A node can have either a matrix or any combination of translation/rotation/scale (TRS) properties.
<https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#node>

Reply:
Understood. This is a nice improvement for not writing default values.

Regarding the builders: It would probably be easier to sort out the JSON first. An early/WIP review would be good.